### PR TITLE
Refactoring of action mailers

### PR DESCRIPTION
**Pull request by eduardoj**
- Organize common parts of action mailers on an application mailer:
   - This is the recommended way of organizing mailers proposed by the Action Mailer Basics guide: https://guides.rubyonrails.org/action_mailer_basics.html
   - Solve the `Rails/ApplicationMailer` RuboCop offenses
- Get rid of this unneded assignment: `@configuration = ::Configuration.fetch`
- Prepare for a further refactoring to get rid of the `@host` variable and use `action_mailer.default_url_options` instead.

### CODE REVIEW

**Comment by danidoni** [comment]
I miss the "why" for this change. Could you add a bit of description explaining what are you trying to improve with this refactoring?

**Comment by eduardoj** [comment]
> I miss the "why" for this change. Could you add a bit of description explaining what are you trying to improve with this refactoring?

Sure. It looked pretty obvious to me:
- Organize common parts of action mailers on an application mailer:
   - This is the recommended way of organizing mailers proposed by the Action Mailer Basics guide: https://guides.rubyonrails.org/action_mailer_basics.html
   - Solve the Rails/ApplicationMailer RuboCop offenses
- Get rid of this unneded assignment: `@configuration = ::Configuration.fetch`
- Prepare for a further refactoring to get rid of the @host variable and use `action_mailer.default_url_options` instead instead.

Basically, do some cleaning, remove duplication of the code and prepare for further refactorings. I added the above points to the description of the pull request.

**Review by danidoni — APPROVED** [review]
